use-cases/client-server-msgs-perf/svmsg_file_server.c

```diff
@@ -293,41 +305,55 @@ main(int argc, char *argv[])
             resp.counter = ++clientp->client_ctr;   // Do Increment
             clientp->num_ops++;                     // Track # of operations
 
-            if (clock_gettime(clock_id, &ts1)) {    // Timing ends
+            if (clock_gettime(tmp_clock_id, &ts1)) { // Timing ends
                 errExit("clock_gettime-ts1");
             }
 
             nsec0 = timespec_to_ns(&ts0);
             nsec1 = timespec_to_ns(&ts1);
-            elapsed_ns = (nsec1 - nsec0);           // Elapsed-ns for this op
+            elapsed_cpu_ns = (clientp->realtime_clock
+                                ? 0
+                                : (nsec1 - nsec0));// Elapsed-ns for this op
 
 #if L3_ENABLED
 
-            // Record time-consumed. (NOTE: See clarification below.)
+            // Record time-consumed. (NOTE: We are alternating between cpu-
+            // clock and real-time clock timing. So, every alternate log-message
+            // will have 0 for this metric. Don't be fooled by it.)
 #  if L3_FASTLOG_ENABLED
             l3_log_fast("Server msg: Increment: ClientID=%d, "
                         "Thread-CPU-time=%" PRIu64 " ns. (L3-fast-log)",
-                        resp.clientId, elapsed_ns);
+                        resp.clientId, elapsed_cpu_ns);
 #  else
             l3_log("Server msg: Increment: ClientID=%d, "
                    "Thread-CPU-time=%" PRIu64 " ns.",
-                   resp.clientId, elapsed_ns);
+                   resp.clientId, elapsed_cpu_ns);
 #  endif
 
+#else
+            // Dead-code to avoid compiler set-but-unsed error for baseline.
+            if (0) { elapsed_cpu_ns++; }
+
 #endif // L3_ENABLED
 
             // Reuse variable to reacquire current TS for accumulation.
             // Cumulative time-consumed gives a read into impact of any
             // logging call that might be done in diff micro-benchmarks.
 
-            if (clock_gettime(clock_id, &ts1)) {    // Cumulative timing ends
-                errExit("clock_gettime-ts1");
+            if (clientp->realtime_clock) {
+                if (clock_gettime(tmp_clock_id, &ts1)) {    // Cumulative real-timing ends
+                    errExit("clock_gettime-ts1");
+                }
+                nsec1 = timespec_to_ns(&ts1);
+                uint64_t elapsed_realtime_ns = (nsec1 - nsec0);
+
+                // Cumulative real-timeelapsed-ns
+                clientp->cumu_real_time_ns += elapsed_realtime_ns;
             }
-            nsec1 = timespec_to_ns(&ts1);
-            elapsed_ns = (nsec1 - nsec0);
-            clientp->cumu_time_ns += elapsed_ns;    // Cumulative elapsed-ns
 
+            clientp->realtime_clock ^= 1;
```

Comment:
what is this doing?

Reply:
ignore it, Greg -- That's an outdated diff, from a prior version of attempts to use diff clocks for timing. This line is now deleted.